nrf_security/Kconfig

```diff
@@ -350,99 +493,219 @@ config VANILLA_MBEDTLS_DHM_C
+	bool "Enable Koblitz curve secp224k1"
+	default n
+	help
+	  MBEDTLS_ECP_DP_SECP224K1_ENABLED setting in mbed TLS config file.
+
+config MBEDTLS_ECP_DP_SECP256K1_ENABLED
+	bool "Enable Koblitz curve secp256k1"
+	default n
+	help
+	  MBEDTLS_ECP_DP_SECP256K1_ENABLED setting in mbed TLS config file.
+
+config MBEDTLS_ECP_DP_BP256R1_ENABLED
+	bool "Enable Brainpool curve bp256r1"
+	depends on (VANILLA_MBEDTLS_ECDSA_C || VANILLA_MBEDTLS_ECP_C || VANILLA_MBEDTLS_ECDH_C  )
+	default n
+	help
+	  MBEDTLS_ECP_DP_BP256R1_ENABLED setting in mbed TLS config file.
+
+config MBEDTLS_ECP_DP_BP384R1_ENABLED
+	bool "Enable Brainpool curve bp384r1"
+	depends on (VANILLA_MBEDTLS_ECDSA_C || VANILLA_MBEDTLS_ECP_C || VANILLA_MBEDTLS_ECDH_C  )
+	default n
+	help
+	  MBEDTLS_ECP_DP_BP384R1_ENABLED setting in mbed TLS config file.
+
+config MBEDTLS_ECP_DP_BP512R1_ENABLED
+	bool "Enable Brainpool curve bp384r1"
+	depends on (VANILLA_MBEDTLS_ECDSA_C || VANILLA_MBEDTLS_ECP_C || VANILLA_MBEDTLS_ECDH_C  )
+	default n
+	help
+	  MBEDTLS_ECP_DP_BP512R1_ENABLED setting in mbed TLS config file.
+
+config MBEDTLS_ECP_DP_CURVE25519_ENABLED
+	bool "Enable Curve25519 aka Edwards25519"
+	default n
+	help
+	  MBEDTLS_ECP_DP_CURVE25519_ENABLED setting in mbed TLS config file.
+
+config MBEDTLS_ECP_DP_CURVE448_ENABLED
+	bool "Enable Curve448 aka Edwards448"
+	depends on (VANILLA_MBEDTLS_ECDSA_C || VANILLA_MBEDTLS_ECP_C || VANILLA_MBEDTLS_ECDH_C	)
+	default n
+	help
+	  MBEDTLS_ECP_DP_CURVE448_ENABLED setting in mbed TLS config file.
+
+endmenu # ECC curves
+
+endmenu # ECC - Elliptic Curve Cryptography
+
+menu "RSA   - Rivest–Shamir–Adleman cryptosystem"
+	depends on (CC310_BACKEND || MBEDTLS_VANILLA_BACKEND)
 
 config MBEDTLS_RSA_C
 	bool
 	prompt "RSA   - Rivest–Shamir–Adleman cryptosystem"
-	depends on (CC310_BACKEND || MBEDTLS_VANILLA_BACKEND)
 	default y
+	depends on CC310_BACKEND || MBEDTLS_VANILLA_BACKEND
 	help
 	  Enable RSA cryptosystem support.
 	  MBEDTLS_RSA_C setting in mbed TLS config file.
 
 choice
 	bool
-	prompt "Backend selection" if MBEDTLS_RSA_C
+	prompt "RSA backend selection" if MBEDTLS_RSA_C
 	depends on NRF_CRYPTO_BACKEND_COMBINATION_0
+	depends on MBEDTLS_RSA_C
 	default CC310_MBEDTLS_RSA_C
 
-	config CC310_MBEDTLS_RSA_C
-	    bool "cc310" if CC310_BACKEND
+config CC310_MBEDTLS_RSA_C
+	bool "cc310" if CC310_BACKEND
+
+config VANILLA_MBEDTLS_RSA_C
+	bool "mbed TLS vanilla" if MBEDTLS_VANILLA_BACKEND
 
-	config VANILLA_MBEDTLS_RSA_C
-	    bool "mbed TLS vanilla" if MBEDTLS_VANILLA_BACKEND
 endchoice
 
-menuconfig HASH_FUNCTIONS_CONFIGURATION
-	bool
-	prompt "SHA   - Secure Hash Algorithm"
-	depends on (CC310_BACKEND || MBEDTLS_VANILLA_BACKEND)
+config MBEDTLS_PKCS1_V15
+	bool "PKCS1 v1.5 support"
+	depends on MBEDTLS_RSA_C
+
+config MBEDTLS_PKCS1_V21
+	bool "PKCS1 v2.1 support"
+	depends on MBEDTLS_RSA_C
 	default y
 
-if HASH_FUNCTIONS_CONFIGURATION
+endmenu # RSA   - Rivest–Shamir–Adleman cryptosystem
+
+menu "SHA   - Secure Hash Algorithm"
```

Comment:
I think changing this from menuconfig to menu is a step back in usability, as users must now enter the submenu to see if the group is configured or not.
The menuconfig allows user to completely disable SHA if it is not needed (SHA as a group will most likely always be needed, but all menu should share look-n-feel).

Reply:
Fixed after some discussion